addons/crm/models/crm_lead.py

```diff
@@ -587,17 +571,16 @@ def _compute_partner_email_update(self):
         for lead in self:
             will_write_email = False
             if lead.partner_id:
-                # beware: void user input gives '' which is different from False
-                lead_email_normalized = tools.email_normalize(lead.email_from) or (lead.email_from if lead.email_from else False)
-                partner_email_normalized = tools.email_normalize(lead.partner_id.email) or lead.partner_id.email
-                will_write_email = lead_email_normalized != partner_email_normalized if lead.partner_id else False
+                lead_email_normalized = tools.email_normalize(lead.email_from) or lead.email_from or False
```

Comment:
Would be good to mention in docstring that when lead or partner emails are "Falsy" (e.g: empty string), we force False to be able to compare False to False
As comparing '' to False returns True, meaning we would synchronize with the partner when we should not.